dashboard/client/src/common/MultiTabLogViewer.tsx

```diff
@@ -30,17 +31,42 @@ export type MultiTabLogViewerTabDetails = {
 export type MultiTabLogViewerProps = {
   tabs: MultiTabLogViewerTabDetails[];
   otherLogsLink?: string;
+  /**
+   * If set, this multi-tab log viewer will remember the last selected tab and start on that tab
+   * the next time this component is rendered.
+   *
+   * Different string values to provide different contexts for this memory. For example, if you
+   * want all multi-tab log viewers in the actor detail page to share one memory, they should have
+   * the same string value here.
+   */
+  tabMemoryContextKey?: string;
 } & ClassNameProps;
 
 export const MultiTabLogViewer = ({
   tabs,
   otherLogsLink,
+  tabMemoryContextKey,
   className,
 }: MultiTabLogViewerProps) => {
   const classes = useStyles();
-  const [value, setValue] = useState(tabs[0]?.title);
+
+  const [rememberedTab, setRememberedTab] = useLocalStorage(
+    `MultiTabLogViewer-tabMemory-${tabMemoryContextKey}`,
+    null,
+  );
+
+  const [value, setValue] = useState(
+    tabMemoryContextKey && rememberedTab ? rememberedTab : tabs[0]?.title,
+  );
   const [expanded, setExpanded] = useState(false);
 
+  useEffect(() => {
+    // If current tab value is not valid, reset to first tab.
+    if (!tabs.some((tab) => tab.title === value)) {
```

Comment:
This is a shared component that accepts tab as props from outside the component.
However, the currently set tab value is controlled by state inside the component. It's possible the caller will send in a new list of tabs and we should handle this situation gracefully. In practice, this doesn't happen today but since this is a shared component, someone can use it differently in the future.